lib/pure/asyncdispatch.nim

```diff
@@ -1228,14 +1228,19 @@ else:
     let newLength = max(len(curList), InitCallbackListSize)
     var newList = newSeqOfCap[Callback](newLength)
 
+    var eventsExtinguished = false
     for cb in curList:
+      if eventsExtinguished:
+        newList.add(cb)
```

Comment:
Don't you need the same `if not cb(fd)` check here?
a previous setting of the `eventsExtinguished` flag from a possibly different cb doesn't guarantee this cb wants to be re-added to the list.

Reply:
It's not that it wants to be re-added, it's that it hasn't been run. It needs to be run when the FD becomes writeable again.